autoload/lsp/lspserver.vim

```diff
@@ -792,19 +792,37 @@ enddef
 # LSP line and column numbers start from zero, whereas Vim line and column
 # numbers start from one. The LSP column number is the character index in the
 # line and not the byte index in the line.
-def GetLspPosition(): dict<number>
+def GetLspPosition(find_ident: bool): dict<number>
   var lnum: number = line('.') - 1
   var col: number = charcol('.') - 1
+  var line = getline('.')
+
+  if find_ident
+    # 1. skip to start of identifier
+    while line[col] != '' && line[col] !~ '\k'
```

Comment:
I think a check is needed to make sure "col" is less then line->len().

Reply:
I think that `line[col] != ''` takes care of it, as `'abc'[123456] == ''`

Reply:
Yes. You are correct.  There is no need to check for the boundary condition here.